examples/step-98/step-98.cc

```diff
@@ -0,0 +1,974 @@
+            // are now stored within each 'phi' object.
+          }
+
+        // Gather the results of the local operator application from the cells
+        // into the 'local_residual' vector for the patch.
+        // 'true' indicates accumulation is needed if multiple cell DoFs map
+        // to the same patch DoF (though typically not the case here).
+        patch_eval.gather_local_to_patch(ArrayView<double>(local_residual),
+                                         true);
+
+        // --- Step 3: Compute the actual residual on the patch ---
+        // residual = rhs - A*solution (Note: local_residual currently holds
+        // A*solution)
+        for (unsigned int i = 0; i < local_residual.size(); ++i)
+          local_residual[i] = local_rhs[i] - local_residual[i];
+
+
+        // --- Step 4: Apply the inverse patch operator ---
+        // Solve the local system A_patch * correction = residual for the
+        // correction. 'inverse' is the PatchTensorInverse object initialized
+        // earlier.
+
+        patch_inverse_operator.apply_inverse(
+          ArrayView<number>(local_correction.data(), n_dofs_interior),
+          ArrayView<const number>(local_residual.data(), n_dofs_interior));
+
+        // inverse.vmult(local_correction, local_residual);
+
+        // --- Step 5: Update the solution ---
+        // The standard smoother update is: u_new = u_old + correction
+        // However, the distribute functions from underlying FEEvaluations
+        // perform an *addition* to the target vector. Since there are DoF that
+        // are shared on multiple cells and we want to add the correction only
+        // once, we need zero out the "duplicates".
+        //
+        // Distribute the patch correction vector back to the individual cells'
+        // FEEvaluation objects. 'false' indicates that DoFs shared beteween
+        // multiple cell are only written the first owning cell, not
+        // accumulated. The values of shared DoFs in subsequent cells will be
+        // set to 0.
+        patch_eval.distribute_patch_to_local(
+          ArrayView<const number>(local_correction), false);
+
+        // Add the corrections stored in the local FEEvaluation objects
+        // to the global solution vector 'dst'.
+        patch_eval.distribute_local_to_global(dst);
+      }
+  }
+
+
+} // namespace Operators
+
+
+namespace LaplaceSolver
+{
+  using namespace dealii;
+
+  /**
+   * Implements a Laplace solver using matrix-free techniques.
+   * There are two main differences between this class and the one from step-37:
+   *  - MatrixFree object for level operators  have to enable evalution on
+   * ghosted cells
+   *  - The smoother is a patch smoother, which is more efficient for high order
+   * elements.  It requires building PatchStorage to store infomration necessary
+   * for loops. The smoother is good enough on its own, there is no need to
```

Comment:
I think this is a good statement. I think it might be interesting to make some small comparison in the results section. I would be very curious to see that - it could be that you would need the MPI implementation to make a good comparison of run time that does not disfavor this class, but in any case the number of iterations and residual evaluations could be very interesting for a reader.